rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/release/ReleaseController.java

```diff
@@ -108,6 +105,24 @@ private Release searchReleaseBySha1(String sha1, User sw360User) throws TExcepti
         return new ResponseEntity<>(halRelease, HttpStatus.OK);
     }
 
+    @RequestMapping(value = RELEASES_URL + "/searchByExternalIds", method = RequestMethod.GET)
+    public ResponseEntity searchByExternalIds(@RequestParam MultiValueMap<String, String> externalIdsMultiMap) throws TException {
```

Comment:
One could refactor that to something like
```java

    @RequestMapping(value = RELEASES_URL + "/searchByExternalIds", method = RequestMethod.GET)
    public ResponseEntity searchByExternalIds(@RequestParam MultiValueMap<String, String> externalIdsMultiMap) throws TException {
        return searchByExternalIds(externalIdsMultiMap,
                releaseService::searchByExternalIds,
                restControllerHelper::convertToEmbeddedRelease,
                (oldR, newR) -> newR.setExternalIds(oldR.getExternalIds()));
    }

    public <T> ResponseEntity searchByExternalIds(MultiValueMap<String, String> externalIdsMultiMap,
                                                  Function<Map<String, Set<String>>, Set<T>> searchByExternalIds,
                                                  Function<T,T> convertToEmbedded,
                                                  BiConsumer<T,T> keepExternalIds) {
        Map<String, Set<String>> externalIds = restControllerHelper.getExternalIdsFromMultiMap(externalIdsMultiMap);
        Set<T> sw360Objects = searchByExternalIds.apply(externalIds);
        List<Resource> releaseResources = new ArrayList<>();

        sw360Objects.forEach(sw360Object -> {
            T embedded = convertToEmbedded.apply(sw360Object);
            keepExternalIds.accept(sw360Object, embedded);
            Resource<T> releaseResource = new Resource<>(embedded);
            releaseResources.add(releaseResource);
        });

        Resources<Resource> resources = new Resources<>(releaseResources);
        return new ResponseEntity<>(resources, HttpStatus.OK);
    }
```

as further improvements (and since the code above currently does not compile) one could add an interface `AwareOfExternalIdsService<Release>` to `Sw360ReleaseService`. This (together with the other comments) would allow to modify the signature of added method to something like
```
    public <T> ResponseEntity searchByExternalIds(MultiValueMap<String, String> externalIdsMultiMap,
                                                  AwareOfExternalIdsService<T> service,
                                                  RestControllerHelper<T> helper) {
```

This method would work for releases, components and projects.